admin/src/utils/wrapWithErrorHandling.js

```diff
@@ -0,0 +1,12 @@
+export default cb => (effects, ...args) => {
+    try {
+        const result = cb(effects, ...args);
```

Comment:
you can wrap cb(effects, ...args) in a Promise.resolve and avoid the test for then